### PR TITLE
docs: update strategic merge patch documentation

### PR DESCRIPTION
**Pull request by slashpai**
Related-to #6080

## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Review by simonpasquier — COMMENTED** [review]
Thanks! Maybe we should revisit the page structure?

1. How to patch a container probe
1. How to inject an environment variable in an existing container
1. How to inject a sidecar container in a custom resource

**Comment by slashpai** [comment]
@simonpasquier Please review again, updated

File: Documentation/user-guides/strategic-merge-patch.md [hunk 1]
```diff
@@ -67,3 +70,46 @@ spec:
...
+spec:
+  containers:
+  - name: "prometheus"
+    env:
+    - name: GOMEMLIMIT
+      value: 6Gi
+```
+
+## How to inject a sidecar container in a custom resource
+
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:95]:
```suggestion
The following manifest injects an additional container to the generated StatefulSet:
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 2]
```diff
@@ -67,3 +70,46 @@ spec:
...
+The following manifest inject the environment variable `GOMEMLIMIT` to the Prometheus container:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  labels:
+    prometheus: self
+  name: self
+  namespace: default
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:85]: We can shorten the manifest and give the resource a more specific name:
```suggestion
metadata:
  name: inject-env-var
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 3]
```diff
@@ -67,3 +70,46 @@ spec:
...
+      value: 6Gi
+```
+
+## How to inject a sidecar container in a custom resource
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  labels:
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:100]: same remark for a shorten version

**Comment by slashpai** [comment]
addressed comments and also made all examples shortened version as well.

**Review by simonpasquier — APPROVED** [review]

**Comment by simonpasquier** [comment]
https://github.com/prometheus-operator/prometheus-operator/pull/5988 creates a small problem because the e2e and unit tests are required to merge. It's not blocking since we can manually override but I'm not sure what is the fix for this (can we say that these jobs are always successful when only .md files are updated?).

image

**Comment by slashpai** [comment]
> https://github.com/prometheus-operator/prometheus-operator/pull/5988 creates a small problem because the e2e and unit tests are required to merge. It's not blocking since we can manually override but I'm not sure what is the fix for this (can we say that these jobs are always successful when only .md files are updated?).
> 
> 
> 
> image
> 
> 

I will check the actions documentation and figure out

**Pull request merged**